traffic_ops/traffic_ops_golang/api/api.go

```diff
@@ -410,8 +410,11 @@ func SendMail(to rfc.EmailAddress, msg []byte, cfg *config.Config) (int, error,
 	if !cfg.SMTP.Enabled {
 		return http.StatusInternalServerError, nil, errors.New("SMTP is not enabled; mail cannot be sent")
 	}
-	auth := smtp.PlainAuth("", cfg.SMTP.User, cfg.SMTP.Password, strings.Split(cfg.SMTP.Address, ":")[0])
-	err := smtp.SendMail(cfg.SMTP.Address, auth, cfg.ConfigTO.EmailFrom.String(), []string{to.String()}, msg)
+	var auth smtp.Auth
+	if cfg.SMTP.User != "" {
+		auth = LoginAuth("", cfg.SMTP.User, cfg.SMTP.Password, strings.Split(cfg.SMTP.Address, ":")[0])
+	}
+	err := smtp.SendMail(cfg.SMTP.Address, auth, cfg.ConfigTO.EmailFrom.Address.Address, []string{to.Address.Address}, msg)
```

Comment:
Does a nil `smtp.Auth` var work with `smtp.SendMail` still? I.e. smtp is enabled, but the user == `""`?

Reply:
yep it does! The SMTP server we tested with actually doesnt have auth so we did exactly that